### DISCUSSION TITLE
How to get mysqld.sock be in the pod environment without a volume?

### DISCUSSION THREAD

**Discussion by DevDorrejo** [containers/podman#22129] [Q&A]
Hello,

I am trying to setup koha-community in a container for a library, but while trying to connect to the database while everything is in a pod, it will fail if try to connect with the next structure: `mysql -h db_koha -u koha -p`, When executed, it will fail asking for mysqld.socket.

So, how can i expose mysqld.socket to the pod?

i run mariadb + temporal ubuntu container.

Thank you.

**Comment by baude** [comment]
consider `--share` when creating the pod

**Reply by DevDorrejo** [reply]
thank for the reply and sorry for bothering you, but can you be a bit more clear pls, i read the man and find that with share is to delimit the namespaces:
(`--share string                             A comma delimited list of kernel namespaces the pod will share (default "ipc,net,uts"`)

But what namespaces you refer which certain I ignore it's existence.

thank you.

**Comment by rhatdan** [comment]
If you want to share mysql from the host into the container, then you need to use a volume.  If you want to have all containers within the Pod see the same mysql socket, then  you have to place the socket on a shared volume.

**Reply by DevDorrejo** [reply]
The only way is adding this to the pod/continaer `-v mysql_socket:/var/run/mysqld/mysqld.sock`?

**Reply by eriksjolund** [reply]
Maybe some day in the future it will be possible to use the `OpenFile=` systemd directive?
I created a [feature request](https://jira.mariadb.org/browse/CONC-652) for __MariaDB Connector/C__
See also
* https://github.com/eriksjolund/mariadb-podman-socket-activation/discussions/10

**Reply by DevDorrejo** [reply]
That's nice, but would not work, because the rootfs share between containers.

**Reply by eriksjolund** [reply]
The Unix socket file does not need to be on the rootfs of the container. I wrote an example with a web server running  on the host that is listening on Unix socket.  A modified curl running in a container fetches a web page from that web server. It works because __systemd__ on the host connects to the Unix socket. Then __systemd__, __podman__ and __conmon__ passes the file descriptor to the container via fork/exec inheritance.

See example:
https://github.com/eriksjolund/podman-OpenFile
